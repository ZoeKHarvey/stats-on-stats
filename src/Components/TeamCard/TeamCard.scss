.team-card {
  border: 2px solid red;
  width: 20%;
  display: flex;
  margin-right: 3%;
  margin-top: 1%;
  flex-direction: column;
  height: 32vh;
  background: rgba(white, .2);
  border: 2px solid rgba(white, .7);
  justify-content: center;
  text-align: center;
}

h4 {
  margin: 0px;
}