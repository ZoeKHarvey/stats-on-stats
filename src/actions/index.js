export const getTeams = (teams) => ({
  type:'GET_TEAMS',
  teams
})

export const getStandings = (standings) => ({
  type: 'GET_STANDINGS',
  standings
})

export const getRoster = (roster) => ({
  type: 'GET_ROSTER',
  roster
})

export const getPlayer = (player) => ({
  type: 'GET_PLAYER',
  player
})

export const getPlayerStats = (player) => ({
  type: 'GET_PLAYER_STATS',
  player
})

export const getTeamSchedule = (schedule) => ({
  type: 'GET_TEAM_SCHEDULE',
  schedule
})

export const getPlayerProjections = (player) => ({
  type: 'GET_PLAYER_PROJECTIONS',
  player
})

export const favoriteTeams = (newTeam) => ({
  type: 'GET_FAVORITE_TEAMS',
  newTeam
})

export const isLoading = (isLoading) => ({
  type: 'IS_LOADING',
  isLoading
})

export const hasError = (errorMsg) => ({
  type: 'HAS_ERROR',
  errorMsg
})



