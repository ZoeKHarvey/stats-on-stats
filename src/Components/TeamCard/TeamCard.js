import React, {Component} from 'react'
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import './TeamCard.scss';
import { getTeams, isLoading, hasError, getRoster, getPlayer, getTeamSchedule } from '../../actions';
import { fetchTeams, fetchRoster, fetchPlayer, fetchPlayerStats, fetchTeamSchedule, fetchPlayerProjections } from '../../apiCalls';
import { Route } from 'react-router-dom';

export class TeamCard extends Component {

  getSingleRoster = async(id) =>{
    const { getRoster } = this.props;
    try {
      const roster = await fetchRoster(id);
      getRoster(roster)
    } catch(error) {
      return 'Roster Not Available'
    }
  }

  getSingleTeamSchedule = async(id) => {
    const { getTeamSchedule } = this.props;
    try {
      const schedule = await fetchTeamSchedule(id);
      getTeamSchedule(this.cleanUpSchedule(schedule))
    } catch(error) {
      return 'No Games Today'
    }
  }

  cleanUpSchedule = (schedule) => {
    return {
      away: {
      awayTeam: schedule.teams.away.team.name,
      awayWins: schedule.teams.away.leagueRecord.wins,
      awayLosses: schedule.teams.away.leagueRecord.losses,
      awayOT: schedule.teams.away.leagueRecord.ot },
      home: {
        homeTeam: schedule.teams.home.team.name,
        homeWins: schedule.teams.home.leagueRecord.wins,
        homeLosses: schedule.teams.home.leagueRecord.losses,
        homeOT: schedule.teams.home.leagueRecord.ot
    }
  }
}

  handleDetails = (e, id) => {
    this.getSingleRoster(id)
    this.getSingleTeamSchedule(id)
    // this.favoriteTeam(e, id)
  }

  render() {
    const {id, name, venue, firstYearOfPlay, division, conference, officialSiteUrl } = this.props;
    // const teamSchedError = teamSchedule || 'No Games Today'
      return (
         <section className="team-card" > 
          <h2>{name}</h2>
          <a href={officialSiteUrl}>Official Site</a>
          <h4>HomeTown: {venue.city}</h4>
          <h4>Founded: {firstYearOfPlay}</h4>
          <h4>Stadium: {venue.name}</h4>
          <h4>Division: {division.name}</h4>
          <h4>Conference: {conference.name}</h4>
          <Link to='/roster' >
            <button className="teamcard__button--details" onMouseEnter={(e) => this.handleDetails(e, id)}>Show Details</button>
          </Link>
        </section>
      )
}   }


export const mapStateToProps = (state) => ({
  teams: state.teams,
  errorMsg: state.errorMsg,
  teamSchedule: state.teamSchedule
})

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    getRoster,
    getTeamSchedule,
  }, dispatch)
)
export default connect(mapStateToProps, mapDispatchToProps)(TeamCard);