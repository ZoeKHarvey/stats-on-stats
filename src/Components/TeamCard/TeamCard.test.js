import React from 'react';
import { shallow } from 'enzyme';
import { TeamCard, mapStateToProps, mapDispatchToProps } from './TeamCard';
import { getRoster, getTeamSchedule } from '../../actions/index';

describe('TeamCard', () => {
  let wrapper;

  let venue = {
    city: 'Philly',
    name: 'Wells Fargo'
  }

  let division = {
    name: 'Eastern'
  }

  let conference = {
    name: 'Metropolitan'
  }

beforeEach(() => {
  wrapper = shallow(
  <TeamCard venue={venue}
            division={division} 
            conference={conference}/>
  );
});
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mockStateToProps', () => {
  it('should return an object with data', () => {
    const mockState = {
      teams: [{}, {}, {}],
      errorMsg: 'error'
    };
    const expected = {
      teams: [{}, {}, {}],
      errorMsg: 'error'
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

  const mockDispatch = jest.fn();

  it('should call dispatch with a getRoster action when handleDetails is clicked', () => {
    let roster = [{}, {}, {}]

    const actionToDispatch = getRoster(roster);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getRoster(roster);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });

  it('should call dispatch with a getTeamSchedule action when handleDetail is called', () => {
    let schedule = {home: 'here', away: 'there'}

    const actionToDispatch = getTeamSchedule(schedule);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getTeamSchedule(schedule);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });

})