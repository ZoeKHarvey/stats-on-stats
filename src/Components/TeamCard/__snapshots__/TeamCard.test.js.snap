// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamCard should match the snapshot 1`] = `ShallowWrapper {}`;
